### ISSUE TITLE
Plugins may only work from $PATH (on macOS)

### ISSUE THREAD

**Issue by jdub** [hashicorp/waypoint#691]
**Describe the bug**
[Installing the plugin](https://www.waypointproject.io/docs/extending-waypoint/creating-plugins/compiling#installing-the-plugin) suggests a few locations to put plugin executables. The easy ones to use on macOS don't appear to work, but putting my plugin executable in `$PATH` does.

**Steps to Reproduce**
* I put my executable in the same directory as an example `waypoint.hcl` file:
```
$ waypoint init
✓ Configuration file appears valid
✓ Connection to Waypoint server was successful
✓ Project "fargate" and all apps are registered with the server.
❌ Failed to load and validate plugins!

! This validation check ensures that you have all the required plugins available
  and the configuration for each plugin (if it exists) is valid. The error message
  below should tell you which plugin(s) failed.

! exec: "waypoint-plugin-cloudformation": executable file not found in $PATH
! Project had errors during initialization.
  Waypoint experienced some errors during project initialization. The output
  above should contain the failure messages. Please correct these errors and
  run 'waypoint init' again.
```

* Not sure what `<waypoint_app_folder>` is, maybe `$(dirname $0)`? I don't want to put my crappy experimental plugins into brew's `/usr/local/bin` just yet. 😆 

* May as well try the `XDG_CONFIG_HOME` option:
```
$ cp waypoint-plugin-cloudformation ~/.config/waypoint/plugins/
$ XDG_CONFIG_HOME=$HOME/.config waypoint init
✓ Configuration file appears valid
✓ Local mode initialized successfully
✓ Project "fargate" and all apps are registered with the server.
❌ Failed to load and validate plugins!

! This validation check ensures that you have all the required plugins available
  and the configuration for each plugin (if it exists) is valid. The error message
  below should tell you which plugin(s) failed.

! exec: "waypoint-plugin-cloudformation": executable file not found in $PATH
! Project had errors during initialization.
  Waypoint experienced some errors during project initialization. The output
  above should contain the failure messages. Please correct these errors and
  run 'waypoint init' again.
```

* But if I copy it to `$HOME/.local/bin`, which is in my `PATH`:
```
$ cp waypoint-plugin-cloudformation ~/.local/bin/
$ waypoint init
✓ Configuration file appears valid
✓ Connection to Waypoint server was successful
✓ Project "fargate" and all apps are registered with the server.
✓ Plugins loaded and configured successfully
✓ Authentication requirements appear satisfied.

Project initialized!

You may now call 'waypoint up' to deploy your project or
commands such as 'waypoint build' to perform steps individually.
```

* Hooray right? But here's where it gets really weird. If I then _remove_ the executable in the current directory, leaving only the one in my `PATH`:
```
$ rm waypoint-plugin-cloudformation
$ which waypoint-plugin-cloudformation
/Users/jeffwaugh/.local/bin/waypoint-plugin-cloudformation
$ waypoint init
✓ Configuration file appears valid
✓ Connection to Waypoint server was successful
✓ Project "fargate" and all apps are registered with the server.
❌ Failed to load and validate plugins!

! This validation check ensures that you have all the required plugins available
  and the configuration for each plugin (if it exists) is valid. The error message
  below should tell you which plugin(s) failed.

! 1 error occurred:
  	* plugin "cloudformation" not found


! Project had errors during initialization.
  Waypoint experienced some errors during project initialization. The output
  above should contain the failure messages. Please correct these errors and
  run 'waypoint init' again.
```

**Expected behavior**
* Using the executable in the same directory would be ideal, particularly for testing and CI/CD use cases.
* It'd be great if `XDG_CONFIG_HOME` fell back to `$HOME/.config` by default on macOS, otherwise I'm not sure there's a consistent location for plugins to live.

**Additional context**
I'm using Waypoint v0.1.3 (cf5706e9) on macOS, installed via brew.

**Comment by tomhjp** [comment]
I ran into the same issue with regards to installing to the working directory. I've raised #695, which works for me locally. Would be great to get some additional testing/input if you fancy trying it out.

The last case is interesting, and I think still part of the same bug. Waypoint finds the plugin in the current directory, then tries to execute in a way that searches $PATH. So when it's in both locations, it finds one and executes the other. And when it's only available locally, it finds that but then can't find a matching executable in $PATH.
